source/tests/utilities/tcp_echo_server.h

```diff
@@ -0,0 +1,161 @@
+#include <cstring>
+#include <memory>
+#include <string>
+#include <thread>
+#ifdef _WIN32
+#include <winsock2.h>
+#else
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <unistd.h>
+#endif
+
+#ifdef _WIN32
+typedef int socklen_t;
+#else
+typedef int SOCKET;
+#endif
+
+class TcpEchoSession : public std::enable_shared_from_this<TcpEchoSession> {
+public:
+    TcpEchoSession(SOCKET socket_fd)
+    : socket_fd_(socket_fd)
+    , stop_session_(false)
+    {
+    }
+
+    ~TcpEchoSession() {
+        stop();
+#ifdef _WIN32
+        closesocket(socket_fd_);
+#else
+        close(socket_fd_);
+#endif
+    }
+
+    void start()
+    {
+        while (!stop_session_) {
+            do_read();
+        }
+    }
+
+    void stop()
+    {
+	    stop_session_ = true;
```

Comment:
Nit: `stop_session_ = true;` is indented too much here.

We don't enforce it as much as python formatting but you can run clang-format on this file if you want. We have the .clang-format file specified here and instructions on setting it up to format on save in VS Code if you want to do that in this [Wiki section](https://github.com/ni/grpc-device/wiki/Development-Tips-and-Tricks#clang-format-on-save).

Reply:
Yeah, I did this change on Ubuntu which seems have weird indention configuration in its vim. Fixed.